### PR TITLE
read packages from packages.config

### PR DESCRIPTION
**Pull request by notofug**
Attempt to solve https://github.com/sensslen/nuget-license/issues/15

Might be too naive but seem to work OK for initial test, please have a look and tell me what you think.

### CODE REVIEW

File: src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs [hunk 1]
```diff
@@ -27,9 +29,26 @@ public IEnumerable<PackageIdentity> GetInstalledPackages(string projectPath, boo
...
             }
 
-            return GetInstalledPackagesFromAssetsFile(includeTransitive, project);
+            if (project.IsPackageReferenceProject())
+                return GetInstalledPackagesFromAssetsFile(includeTransitive, project);
+
+            return GetInstalledPackagesFromPackagesConfig(project);
+        }
+
+        private IEnumerable<PackageIdentity> GetInstalledPackagesFromPackagesConfig(IProject project)
```
Comment (sensslen) [inline src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs]: There is a problem with this code: This code does not properly follow transitive dependencies.

**Review by sensslen — CHANGES_REQUESTED** [review]
Unfortunately the code you propose does not work properly. We offer a flag that allows to also print transitive dependencies (see https://devblogs.microsoft.com/nuget/introducing-transitive-dependencies-in-visual-studio/ for more information). To test this feature, can you please add more integration tests to actually test this feature?

Also I would love if you could use as much tooling already available instead of implementing the xml parsing on your own.... --> See https://stackoverflow.com/questions/33035704/how-to-read-the-list-of-nuget-packages-in-packages-config-programatically for an example

File: src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs (hunk 1 above)
Comment (notofug) [reply src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs]: I was under the impression that special handling of "transitive dependencies" was something that came with PackageReference-format and was very differently handled when using packages.config ; 
With packages.config both direct and transitive dependencies are added at install-time : https://learn.microsoft.com/en-us/nuget/concepts/dependency-resolution#dependency-resolution-with-packagesconfig
So 'packages.config' is a flat list of used packages where all must be considered 'direct dependencies' - or is there something I am missing here?

Comment (notofug) [reply src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs]: Will see if I can get rid of xml-parsing as suggested

Comment (sensslen) [reply src/NuGetUtility/ReferencedPackagesReader/ReferencedPackageReader.cs]: OK, right - I'm sorry. I'm not very familiar with packages.config....

**Review by sensslen — APPROVED** [review]

**Review by sensslen — CHANGES_REQUESTED** [review]
Thank you very much for your contribution. Unfortunately there are currently two issues I see. Would you mind fixing these?

There are formatting issues.....
These can be resolved very easily by running `dotnet format [path to solution]`

Also there is an error `'ManagementClass' is only supported on: 'windows'`

**Comment by notofug** [comment]
> There are formatting issues..... These can be resolved very easily by running `dotnet format [path to solution]`
> Also there is an error `'ManagementClass' is only supported on: 'windows'`

I tried to isolate the use of `ManagementClass` to windows-builds only, which makes sense since msbuild c++ projects are locked to windows anyway as far as I am aware of. In fact this part of the PR deals explicitly with supporting c++ - projects, since such projects are limited to using 'packages.config' - style if using nuget.
Please check if this change fixes the build issue

**Comment by sensslen** [comment]
Ok, so there are still test failures - can you please run the tests locally to fix all of them?

**Comment by notofug** [comment]
I had to move the file-reading to a wrapper-class in order to satisfy the 'architectural test', please check if that makes sense or we should attack it in another way? (Tests seem to pass locally now at least ;-))

**Review by sensslen — APPROVED** [review]
Looks good now

**Pull request merged**